scripts/create-model.py

```diff
@@ -123,122 +126,157 @@ def write(self):
         add a box to the shape
         :param l: depth
-        :type l: double
+        :type l: float
         :param w: width
-        :type w: double
+        :type w: float
         :param h: height
-        :type h: double
+        :type h: float
         :param x: relative position in x
-        :type x: double
+        :type x: float
         :param y: relative position in y
-        :type y: double
+        :type y: float
         :param z: relative position in z
-        :type z: double
+        :type z: float
         :param comment: name of the box
         :type comment: str
         :return: no return
         """
+        assert(isinstance(l, float))
+        assert(isinstance(w, float))
+        assert(isinstance(h, float))
+        assert(isinstance(x, float))
+        assert(isinstance(y, float))
+        assert(isinstance(z, float))
+
         self.data['shape']['compound'].append({'box': {'#': comment, 'pose': {'x': x, 'y': y, 'z': z},
                                                        'size': {'x': l, 'y': w, 'z': h}}})
 
     def add_area(self, name, area_type, data):
+        # type: (str, str, Union[dict, float, list, str]) -> None
         """
         add an area to an object
         :param name: name of the area, e.g. offset, shape
-        :param area_type: str
+        :type area_type: str
         :param data: data of the area
-        :param data: dict/list
+        :type data: Union[dict, float, list, str]
         :return: no return
         """
+        assert(isinstance(name, str))
+        assert(isinstance(area_type, str))
+        assert(isinstance(data, (dict, float, list, str)))
+
         self.data['areas'].append({'name': name,
-                                  area_type: data})
+                                   area_type: data})
 
     def add_in_front_of(self, depth, width, side_clearance=DEFAULT_IN_FRONT_SIDE_CLEARANCE,
                         distance=DEFAULT_IN_FRONT_DISTANCE,
                         size=DEFAULT_IN_FRONT_SIZE):
+        # type: (float, float, float, float, float) -> None
         """
         add an 'in_front_of' area to an object. It takes into account the size of the object
         :param depth: depth of the object
-        :type depth: double
+        :type depth: float
         :param width: width of the object
-        :type width: double
+        :type width: float
         :param side_clearance: The area will be this much smaller on each side than the object
-        :type side_clearance: double
+        :type side_clearance: float
         :param distance: distance between object and area
-        :type distance: double
+        :type distance: float
         :param size: depth of the area
-        :type size: double
+        :type size: float
         :return: no return
         """
-
-        boxmin = {'x': round(depth/2 + distance, ROUND_LEVEL),
-                  'y': round(-width/2 + side_clearance, ROUND_LEVEL),
+        assert(isinstance(depth, float))
```

Comment:
Do these need to be floats? I guess ints should also work, provided that the divisions below are by 2.0